src/models/userMedia.ts

```diff
@@ -405,4 +435,24 @@ export class UserMedia {
     }
     showBroadcast(this.userName, videoTrack, this.audioTrack);
   }
+
+  tryShowScreenShare() {
+    if (!this.screenTrack) return;
+
```

Comment:
Haven't looked through the code enough to know exactly where this comment should go but: you may want to also use the `supportedBrowser` method to see if `supportsScreenShare` is true before trying to screen share too. 

Another thing to consider: you can turn off the screen share option for rooms. You may want to check screen sharing is enabled before actually displaying the screen share button. (Right now, if you try to screen share with a room where it's not enabled, you get a console error but no UI feedback to know what the issue is).

Small thing but would be nice to have :)

Reply:
Ah so I just saw you're already doing this second check mentioned above! but.. something funky is happening. If I join a room where screen sharing is enabled and then leave via the leave button, and then immediately join a room where the screen sharing is not enable, the button still appears. But if I start with the room where it's not enabled, the button doesn't show.

So it looks like the bug is somewhere in resetting things on leave (I would guess).

Reply:
Ah ha good to know, I'll fix that tomorrow (and add the `supportedBrowser` check as well), thanks!